twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -113,9 +117,11 @@ public Integer apply(BitSet input) {
   private final Location applicationLocation;
   private final Set<String> runnableNames;
   private final Map<String, Map<String, String>> logLevels;
+  private final Map<String, Integer> maxRetries;
```

Comment:
Good catch.  Yes.  I missed that.  I will update the pull request with a fix.  When the number of instances increases, `deltaInstances * maxRetries` will be added to this instance adjusted maximum.  When the number of instances decreases, the `deltaInstance*maxRetries` will be subtracted from the instance adjusted maximum.  Additionally upon a decrease, if the number of retries exceeds the instance adjusted maximum it will be set to the instance adjusted maximum.